// @terra-money/feather.js/dist/core/alliance/proposals/MsgDeleteAllianceProposal.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgDeleteAllianceProposal = void 0;
var json_1 = require("../../../util/json");
var any_1 = require("@terra-money/legacy.proto/google/protobuf/any");
var gov_1 = require("@terra-money/terra.proto/alliance/gov");
/**
 *  MsgDeleteAllianceProposal is gov content type to delete an existing alliance
 */
var MsgDeleteAllianceProposal = /** @class */ (function (_super) {
    __extends(MsgDeleteAllianceProposal, _super);
    /**
     * @description Delete an existent alliance using the gov module
     * @param title of the proposal
     * @param description of the proposal
     * @param denom of the asset. It could either be a native token or an IBC token
     */
    function MsgDeleteAllianceProposal(title, description, denom) {
        var _this = _super.call(this) || this;
        _this.title = title;
        _this.description = description;
        _this.denom = denom;
        return _this;
    }
    MsgDeleteAllianceProposal.fromAmino = function (data, _) {
        _;
        data;
        throw new Error('not implemented');
    };
    MsgDeleteAllianceProposal.prototype.toAmino = function (_) {
        _;
        throw new Error('not implemented');
    };
    MsgDeleteAllianceProposal.fromData = function (data, _) {
        _;
        var title = data.title, description = data.description, denom = data.denom;
        return new MsgDeleteAllianceProposal(title, description, denom);
    };
    MsgDeleteAllianceProposal.prototype.toData = function (_) {
        _;
        var _a = this, title = _a.title, description = _a.description, denom = _a.denom;
        return {
            '@type': '/alliance.alliance.MsgDeleteAllianceProposal',
            title: title,
            description: description,
            denom: denom,
        };
    };
    MsgDeleteAllianceProposal.fromProto = function (proto, _) {
        _;
        return new MsgDeleteAllianceProposal(proto.title, proto.description, proto.denom);
    };
    MsgDeleteAllianceProposal.prototype.toProto = function (_) {
        _;
        var _a = this, title = _a.title, description = _a.description, denom = _a.denom;
        return gov_1.MsgDeleteAllianceProposal.fromPartial({
            title: title,
            description: description,
            denom: denom,
        });
    };
    MsgDeleteAllianceProposal.prototype.packAny = function (_) {
        _;
        return any_1.Any.fromPartial({
            typeUrl: '/alliance.alliance.MsgDeleteAllianceProposal',
            value: gov_1.MsgDeleteAllianceProposal.encode(this.toProto()).finish(),
        });
    };
    MsgDeleteAllianceProposal.unpackAny = function (msgAny, _) {
        _;
        return MsgDeleteAllianceProposal.fromProto(gov_1.MsgDeleteAllianceProposal.decode(msgAny.value));
    };
    return MsgDeleteAllianceProposal;
}(json_1.JSONSerializable));
exports.MsgDeleteAllianceProposal = MsgDeleteAllianceProposal;
