// @terra-money/feather.js/dist/core/jax/msgs/MsgStoreCode.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgStoreCode = void 0;
var json_1 = require("../../../util/json");
var Coins_1 = require("../../Coins");
var any_1 = require("@terra-money/terra.proto/google/protobuf/any");
var tx_1 = require("@terra-money/terra.proto/jax/tx");
var MsgStoreCode = /** @class */ (function (_super) {
    __extends(MsgStoreCode, _super);
    /**
     * @param creator contract deployer
     * @param admin address of contract admin, usually contract deployer
     * @param code the JavaScript source code of the contract
     * @param params json stringified params to pass to init
     * @param coins coins to be sent to contract
     */
    function MsgStoreCode(creator, admin, code, params, coins) {
        if (coins === void 0) { coins = {}; }
        var _this = _super.call(this) || this;
        _this.creator = creator;
        _this.admin = admin;
        _this.code = code;
        _this.params = params;
        _this.coins = new Coins_1.Coins(coins);
        return _this;
    }
    MsgStoreCode.fromAmino = function (data, _) {
        var _a = data.value, creator = _a.creator, admin = _a.admin, code = _a.code, params = _a.params;
        return new MsgStoreCode(creator, admin, code, params);
    };
    MsgStoreCode.prototype.toAmino = function (_) {
        var _a = this, creator = _a.creator, admin = _a.admin, code = _a.code, params = _a.params, coins = _a.coins;
        return {
            type: 'jax/MsgStoreCode',
            value: {
                creator: creator,
                admin: admin,
                code: code,
                params: params,
                coins: coins.toAmino(),
            },
        };
    };
    MsgStoreCode.fromProto = function (proto, _) {
        var p = proto;
        return new MsgStoreCode(p.creator, p.admin, p.code, p.params);
    };
    MsgStoreCode.prototype.toProto = function (_) {
        var _a = this, creator = _a.creator, admin = _a.admin, code = _a.code, params = _a.params, coins = _a.coins;
        return tx_1.MsgStoreCode.fromPartial({
            creator: creator,
            admin: admin,
            code: code,
            params: params,
            coins: coins.toProto(),
        });
    };
    MsgStoreCode.prototype.packAny = function (isClassic) {
        return any_1.Any.fromPartial({
            typeUrl: '/jax.MsgStoreCode',
            value: tx_1.MsgStoreCode.encode(this.toProto(isClassic)).finish(),
        });
    };
    MsgStoreCode.unpackAny = function (msgAny, isClassic) {
        return MsgStoreCode.fromProto(tx_1.MsgStoreCode.decode(msgAny.value), isClassic);
    };
    MsgStoreCode.fromData = function (data, _) {
        var _a = data, creator = _a.creator, admin = _a.admin, code = _a.code, params = _a.params, coins = _a.coins;
        return new MsgStoreCode(creator, admin, code, params, Coins_1.Coins.fromData(coins));
    };
    MsgStoreCode.prototype.toData = function (_) {
        var _a = this, creator = _a.creator, admin = _a.admin, code = _a.code, params = _a.params, coins = _a.coins;
        return {
            '@type': '/jax.MsgStoreCode',
            creator: creator,
            admin: admin,
            code: code,
            params: params,
            coins: coins.toData(),
        };
    };
    return MsgStoreCode;
}(json_1.JSONSerializable));
exports.MsgStoreCode = MsgStoreCode;
