// @terra-money/terra.proto/stride/mint/v1beta1/genesis.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const mint_1 = require("../../../stride/mint/v1beta1/mint");
exports.protobufPackage = "stride.mint.v1beta1";
const baseGenesisState = { reductionStartedEpoch: long_1.default.ZERO };
exports.GenesisState = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.minter !== undefined) {
            mint_1.Minter.encode(message.minter, writer.uint32(10).fork()).ldelim();
        }
        if (message.params !== undefined) {
            mint_1.Params.encode(message.params, writer.uint32(18).fork()).ldelim();
        }
        if (!message.reductionStartedEpoch.isZero()) {
            writer.uint32(24).int64(message.reductionStartedEpoch);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseGenesisState);
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.minter = mint_1.Minter.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.params = mint_1.Params.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.reductionStartedEpoch = reader.int64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseGenesisState);
        if (object.minter !== undefined && object.minter !== null) {
            message.minter = mint_1.Minter.fromJSON(object.minter);
        }
        else {
            message.minter = undefined;
        }
        if (object.params !== undefined && object.params !== null) {
            message.params = mint_1.Params.fromJSON(object.params);
        }
        else {
            message.params = undefined;
        }
        if (object.reductionStartedEpoch !== undefined && object.reductionStartedEpoch !== null) {
            message.reductionStartedEpoch = long_1.default.fromString(object.reductionStartedEpoch);
        }
        else {
            message.reductionStartedEpoch = long_1.default.ZERO;
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.minter !== undefined && (obj.minter = message.minter ? mint_1.Minter.toJSON(message.minter) : undefined);
        message.params !== undefined && (obj.params = message.params ? mint_1.Params.toJSON(message.params) : undefined);
        message.reductionStartedEpoch !== undefined &&
            (obj.reductionStartedEpoch = (message.reductionStartedEpoch || long_1.default.ZERO).toString());
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseGenesisState);
        if (object.minter !== undefined && object.minter !== null) {
            message.minter = mint_1.Minter.fromPartial(object.minter);
        }
        else {
            message.minter = undefined;
        }
        if (object.params !== undefined && object.params !== null) {
            message.params = mint_1.Params.fromPartial(object.params);
        }
        else {
            message.params = undefined;
        }
        if (object.reductionStartedEpoch !== undefined && object.reductionStartedEpoch !== null) {
            message.reductionStartedEpoch = object.reductionStartedEpoch;
        }
        else {
            message.reductionStartedEpoch = long_1.default.ZERO;
        }
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
