// @terra-money/feather.js/dist/core/custom/msgs/MsgAminoCustom.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgAminoCustom = void 0;
var json_1 = require("../../../util/json");
/**
 * A delegator can submit this message to send more alliance assets
 * to be staked through the alliance module in a validator.
 */
var MsgAminoCustom = /** @class */ (function (_super) {
    __extends(MsgAminoCustom, _super);
    /**
     *
     * @param delegatorAddress delegator's account address
     * @param validatorAddress validator's operator address
     * @param amount amount of alliance assets to be sent for delegation
     */
    function MsgAminoCustom(aminoMsg) {
        var _this = _super.call(this) || this;
        _this.aminoMsg = aminoMsg;
        return _this;
    }
    MsgAminoCustom.prototype.toAmino = function (_) {
        _;
        return this.aminoMsg;
    };
    MsgAminoCustom.fromAmino = function (data, _) {
        _;
        return new MsgAminoCustom(data);
    };
    MsgAminoCustom.prototype.toProto = function (_) {
        _;
        throw new Error('Protobuf not supported for MsgAminoCustom');
    };
    MsgAminoCustom.prototype.packAny = function (_) {
        _;
        throw new Error('Protobuf not supported for MsgAminoCustom');
    };
    MsgAminoCustom.unpackAny = function (msgAny, _) {
        msgAny;
        _;
        throw new Error('Protobuf not supported for MsgAminoCustom');
    };
    MsgAminoCustom.fromData = function (data, _) {
        _;
        var msg = data.msg;
        return new MsgAminoCustom(msg);
    };
    MsgAminoCustom.prototype.toData = function (_) {
        _;
        return {
            '@type': 'MsgCustomAmino',
            msg: this.aminoMsg,
        };
    };
    return MsgAminoCustom;
}(json_1.JSONSerializable));
exports.MsgAminoCustom = MsgAminoCustom;
