// @terra-money/terra.proto/stride/records/callbacks.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferCallback = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "stride.records";
const baseTransferCallback = { depositRecordId: long_1.default.UZERO };
exports.TransferCallback = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (!message.depositRecordId.isZero()) {
            writer.uint32(8).uint64(message.depositRecordId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseTransferCallback);
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.depositRecordId = reader.uint64();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseTransferCallback);
        if (object.depositRecordId !== undefined && object.depositRecordId !== null) {
            message.depositRecordId = long_1.default.fromString(object.depositRecordId);
        }
        else {
            message.depositRecordId = long_1.default.UZERO;
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.depositRecordId !== undefined &&
            (obj.depositRecordId = (message.depositRecordId || long_1.default.UZERO).toString());
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseTransferCallback);
        if (object.depositRecordId !== undefined && object.depositRecordId !== null) {
            message.depositRecordId = object.depositRecordId;
        }
        else {
            message.depositRecordId = long_1.default.UZERO;
        }
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
