// @opzlab/terrain/lib/commands/new.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const command_1 = require("@oclif/command");
const template_scaffolding_1 = require("@terra-money/template-scaffolding");
const cli_ux_1 = tslib_1.__importDefault(require("cli-ux"));
const path = tslib_1.__importStar(require("path"));
const fs = tslib_1.__importStar(require("fs"));
const child_process_1 = require("child_process");
const case_1 = require("case");
const dedent_1 = tslib_1.__importDefault(require("dedent"));
const axios_1 = tslib_1.__importDefault(require("axios"));
const TerrainCLI_1 = tslib_1.__importDefault(require("../TerrainCLI"));
const config_1 = require("../config");
class New extends command_1.Command {
    async run() {
        const { args, flags } = this.parse(New);
        const appDir = path.join(process.cwd(), flags.path, args.name);
        const contractDir = path.join(appDir, 'contracts', args.name);
        const frontendDir = path.join(appDir, 'frontend');
        if (fs.existsSync(appDir)) {
            throw Error(`Folder '${args.name}' already exists under path '${flags.path}'.\nTip: Use another path or project name`);
        }
        const templateEntries = {
            'project-name': args.name,
            'client-name': `${(0, case_1.pascal)(args.name)}Client`,
            // Crates cannot have dashes, and Rust will map underscores to dashes for us.
            crate_name: args.name.replaceAll('-', '_'),
            authors: flags.authors,
            ' "now" | date: "%Y" ': `${new Date().getFullYear()}`,
        };
        cli_ux_1.default.log(`🚀 Generating app ${args.name}:`);
        cli_ux_1.default.action.start('  🛠  Workspace');
        await template_scaffolding_1.TemplateScaffolding.from({
            remoteUrl: 'https://codeload.github.com/terra-money/terrain-core-template/zip/refs/heads/main',
            subFolder: 'terrain-core-template-main',
            localOptions: {
                folderUrl: appDir,
            },
            replace: {
                entries: templateEntries,
            },
        });
        const res = await axios_1.default.get('https://station-assets.terra.money/chains.json');
        fs.writeFileSync(path.join(appDir, 'config.terrain.json'), JSON.stringify(Object.assign(Object.assign({}, config_1.GLOBAL_CONFIG), res.data), null, 2));
        cli_ux_1.default.action.stop();
        cli_ux_1.default.action.start('  📝 Contract');
        await template_scaffolding_1.TemplateScaffolding.from({
            remoteUrl: `https://codeload.github.com/InterWasm/cw-template/zip/refs/heads/${flags.version}`,
            subFolder: `cw-template-${flags.version}`,
            localOptions: {
                folderUrl: contractDir,
            },
            replace: {
                entries: templateEntries,
            },
        });
        cli_ux_1.default.action.stop();
        cli_ux_1.default.action.start('  💻 Frontend');
        if (flags.framework === 'react') {
            await template_scaffolding_1.TemplateScaffolding.from({
                remoteUrl: 'https://codeload.github.com/terra-money/terrain-frontend-template/zip/refs/heads/main',
                subFolder: 'terrain-frontend-template-main',
                localOptions: {
                    folderUrl: frontendDir,
                },
                replace: {
                    entries: templateEntries,
                },
            });
        }
        else {
            await template_scaffolding_1.TemplateScaffolding.from({
                remoteUrl: 'https://codeload.github.com/terra-money/wallet-provider/zip/refs/heads/main',
                subFolder: `wallet-provider-main/templates/${flags.framework}`,
                localOptions: {
                    folderUrl: frontendDir,
                },
            });
        }
        cli_ux_1.default.action.stop();
        // Install app dependencies.
        process.chdir(appDir);
        cli_ux_1.default.action.start('  🏗  Installing app dependencies');
        await (0, child_process_1.execSync)('npm i --loglevel error', {
            stdio: ['ignore', 'ignore', 'inherit'],
        });
        cli_ux_1.default.action.stop();
        // Install frontend dependencies.
        process.chdir(frontendDir);
        cli_ux_1.default.action.start('  🔧 Installing frontend dependencies');
        await (0, child_process_1.execSync)('npm i --loglevel error', {
            stdio: ['ignore', 'ignore', 'inherit'],
        });
        cli_ux_1.default.action.stop();
        TerrainCLI_1.default.success((0, dedent_1.default) `
      Application "${args.name}" was successfully generated.\n
      Now, you can change into the contract directory:\n
      "cd ${args.name}"\n
      And try to deploy it to your preferred Terra network:\n
      "terrain deploy ${args.name} --signer <signer-wallet> --network" "<desired-network>"\n
      "NOTE:" To deploy your contract to the "LocalTerra" network utilizing the preconfigured test wallet "test1" as the signer, use the following command:\n
      "terrain deploy ${args.name}"
    `, 'Application Generated');
    }
}
New.description = 'Create new dapp from template.';
New.examples = [
    '$ terrain new awesome-dapp',
    '$ terrain new awesome-dapp --path path/to/dapp',
    '$ terrain new awesome-dapp --path path/to/dapp --authors "ExampleAuthor<example@email.domain>"',
    '$ terrain new awesome-dapp --path path/to/dapp --framework vue --authors "ExampleAuthor<example@email.domain>"',
];
New.flags = {
    path: command_1.flags.string({
        description: 'Path to create the workspace',
        default: '.',
    }),
    framework: command_1.flags.string({
        description: 'Choose the frontend framework you want to use. Non-react framework options have better wallet-provider support but less streamlined contract integration.',
        options: ['react', 'vue', 'svelte', 'next', 'vite', 'lit'],
        default: 'react',
    }),
    version: command_1.flags.string({
        default: '1.0',
    }),
    authors: command_1.flags.string({
        default: 'Terra Money <core@terra.money>',
    }),
};
New.args = [{ name: 'name', required: true }];
exports.default = New;
