// @terra-money/feather.js/dist/core/wasm/AbsoluteTxPosition.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbsoluteTxPosition = void 0;
var types_1 = require("@terra-money/terra.proto/cosmwasm/wasm/v1/types");
var json_1 = require("../../util/json");
var long_1 = __importDefault(require("long"));
/**
 *
 */
var AbsoluteTxPosition = /** @class */ (function (_super) {
    __extends(AbsoluteTxPosition, _super);
    /**
     * @param block_height
     * @param tx_index
     */
    function AbsoluteTxPosition(block_height, tx_index) {
        var _this = _super.call(this) || this;
        _this.block_height = block_height;
        _this.tx_index = tx_index;
        return _this;
    }
    AbsoluteTxPosition.fromAmino = function (data) {
        return new AbsoluteTxPosition(Number.parseInt(data.block_height), Number.parseInt(data.tx_index));
    };
    AbsoluteTxPosition.prototype.toAmino = function () {
        var res = {
            block_height: this.block_height.toFixed(),
            tx_index: this.tx_index.toFixed(),
        };
        return res;
    };
    AbsoluteTxPosition.fromData = function (data) {
        return new AbsoluteTxPosition(Number.parseInt(data.block_height), Number.parseInt(data.tx_index));
    };
    AbsoluteTxPosition.prototype.toData = function () {
        var res = {
            block_height: this.block_height.toFixed(),
            tx_index: this.tx_index.toFixed(),
        };
        return res;
    };
    AbsoluteTxPosition.fromProto = function (proto) {
        return new AbsoluteTxPosition(proto.blockHeight.toNumber(), proto.txIndex.toNumber());
    };
    AbsoluteTxPosition.prototype.toProto = function () {
        return types_1.AbsoluteTxPosition.fromPartial({
            blockHeight: long_1.default.fromNumber(this.block_height),
            txIndex: long_1.default.fromNumber(this.tx_index),
        });
    };
    return AbsoluteTxPosition;
}(json_1.JSONSerializable));
exports.AbsoluteTxPosition = AbsoluteTxPosition;
