// @terra-money/feather.js/dist/core/wasm/AccessConfigUpdate.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessConfigUpdate = void 0;
var proposal_1 = require("@terra-money/terra.proto/cosmwasm/wasm/v1/proposal");
var json_1 = require("../../util/json");
var AccessConfig_1 = require("./AccessConfig");
var long_1 = __importDefault(require("long"));
/**
 *
 */
var AccessConfigUpdate = /** @class */ (function (_super) {
    __extends(AccessConfigUpdate, _super);
    /**
     * @param code_id the reference to the stored WASM code to be updated
     * @param instantiate_permission to apply to the set of code ids
     */
    function AccessConfigUpdate(code_id, instantiate_permission) {
        var _this = _super.call(this) || this;
        _this.code_id = code_id;
        _this.instantiate_permission = instantiate_permission;
        return _this;
    }
    AccessConfigUpdate.fromAmino = function (data) {
        return new AccessConfigUpdate(Number.parseInt(data.code_id), data.instantiate_permission
            ? AccessConfig_1.AccessConfig.fromAmino(data.instantiate_permission)
            : undefined);
    };
    AccessConfigUpdate.prototype.toAmino = function () {
        var _a;
        var res = {
            code_id: this.code_id.toFixed(),
            instantiate_permission: (_a = this.instantiate_permission) === null || _a === void 0 ? void 0 : _a.toAmino(),
        };
        return res;
    };
    AccessConfigUpdate.fromData = function (data) {
        return new AccessConfigUpdate(Number.parseInt(data.code_id), data.instantiate_permission
            ? AccessConfig_1.AccessConfig.fromData(data.instantiate_permission)
            : undefined);
    };
    AccessConfigUpdate.prototype.toData = function () {
        var _a;
        var res = {
            code_id: this.code_id.toFixed(),
            instantiate_permission: (_a = this.instantiate_permission) === null || _a === void 0 ? void 0 : _a.toData(),
        };
        return res;
    };
    AccessConfigUpdate.fromProto = function (proto) {
        return new AccessConfigUpdate(proto.codeId.toNumber(), proto.instantiatePermission
            ? AccessConfig_1.AccessConfig.fromProto(proto.instantiatePermission)
            : undefined);
    };
    AccessConfigUpdate.prototype.toProto = function () {
        var _a;
        return proposal_1.AccessConfigUpdate.fromPartial({
            codeId: long_1.default.fromNumber(this.code_id),
            instantiatePermission: (_a = this.instantiate_permission) === null || _a === void 0 ? void 0 : _a.toProto(),
        });
    };
    return AccessConfigUpdate;
}(json_1.JSONSerializable));
exports.AccessConfigUpdate = AccessConfigUpdate;
