// @terra-money/feather.js/dist/core/wasm/msgs/tokenfactory/MsgBurn.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgBurn = void 0;
var json_1 = require("../../../../util/json");
var any_1 = require("@terra-money/terra.proto/google/protobuf/any");
var tx_1 = require("@terra-money/terra.proto/cosmwasm/tokenfactory/v1beta1/tx");
var Coin_1 = require("../../../Coin");
/**
 * MsgBurn is the sdk.Msg type for allowing an admin account to burn
 * a token.  For now, we only support burning from the sender account.
 */
var MsgBurn = /** @class */ (function (_super) {
    __extends(MsgBurn, _super);
    /**
     *
     * @param sender internal account or external sender address
     * @param amount amount of coins to burn
     */
    function MsgBurn(sender, amount) {
        var _this = _super.call(this) || this;
        _this.sender = sender;
        _this.amount = amount;
        return _this;
    }
    MsgBurn.prototype.toAmino = function (_) {
        _;
        var _a = this, sender = _a.sender, amount = _a.amount;
        return {
            type: 'osmosis/tokenfactory/burn',
            value: {
                sender: sender,
                amount: amount,
            },
        };
    };
    MsgBurn.fromProto = function (proto, _) {
        _;
        return new MsgBurn(proto.sender, Coin_1.Coin.fromProto(proto.amount));
    };
    MsgBurn.prototype.toProto = function (_) {
        _;
        var _a = this, sender = _a.sender, amount = _a.amount;
        return tx_1.MsgBurn.fromPartial({
            sender: sender,
            amount: amount === null || amount === void 0 ? void 0 : amount.toProto(),
        });
    };
    MsgBurn.prototype.packAny = function (_) {
        _;
        return any_1.Any.fromPartial({
            typeUrl: '/cosmwasm.tokenfactory.v1beta1.MsgBurn',
            value: tx_1.MsgBurn.encode(this.toProto()).finish(),
        });
    };
    MsgBurn.unpackAny = function (msgAny, _) {
        _;
        return MsgBurn.fromProto(tx_1.MsgBurn.decode(msgAny.value));
    };
    MsgBurn.fromData = function (data, _) {
        _;
        var sender = data.sender, amount = data.amount;
        return new MsgBurn(sender, amount);
    };
    MsgBurn.prototype.toData = function (_) {
        _;
        var _a = this, sender = _a.sender, amount = _a.amount;
        return {
            '@type': '/cosmwasm.tokenfactory.v1beta1.MsgBurn',
            sender: sender,
            amount: amount,
        };
    };
    MsgBurn.fromAmino = function (data) {
        var _a = data.value, sender = _a.sender, amount = _a.amount;
        return new MsgBurn(sender, amount);
    };
    return MsgBurn;
}(json_1.JSONSerializable));
exports.MsgBurn = MsgBurn;
