// @terra-money/terra.proto/cosmos/base/store/v1beta1/listening.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockMetadata_DeliverTx = exports.BlockMetadata = exports.StoreKVPair = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const types_1 = require("../../../../tendermint/abci/types");
exports.protobufPackage = "cosmos.base.store.v1beta1";
const baseStoreKVPair = { storeKey: "", delete: false };
exports.StoreKVPair = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.storeKey !== "") {
            writer.uint32(10).string(message.storeKey);
        }
        if (message.delete === true) {
            writer.uint32(16).bool(message.delete);
        }
        if (message.key.length !== 0) {
            writer.uint32(26).bytes(message.key);
        }
        if (message.value.length !== 0) {
            writer.uint32(34).bytes(message.value);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseStoreKVPair);
        message.key = new Uint8Array();
        message.value = new Uint8Array();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.storeKey = reader.string();
                    break;
                case 2:
                    message.delete = reader.bool();
                    break;
                case 3:
                    message.key = reader.bytes();
                    break;
                case 4:
                    message.value = reader.bytes();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseStoreKVPair);
        message.key = new Uint8Array();
        message.value = new Uint8Array();
        if (object.storeKey !== undefined && object.storeKey !== null) {
            message.storeKey = String(object.storeKey);
        }
        else {
            message.storeKey = "";
        }
        if (object.delete !== undefined && object.delete !== null) {
            message.delete = Boolean(object.delete);
        }
        else {
            message.delete = false;
        }
        if (object.key !== undefined && object.key !== null) {
            message.key = bytesFromBase64(object.key);
        }
        if (object.value !== undefined && object.value !== null) {
            message.value = bytesFromBase64(object.value);
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.storeKey !== undefined && (obj.storeKey = message.storeKey);
        message.delete !== undefined && (obj.delete = message.delete);
        message.key !== undefined &&
            (obj.key = base64FromBytes(message.key !== undefined ? message.key : new Uint8Array()));
        message.value !== undefined &&
            (obj.value = base64FromBytes(message.value !== undefined ? message.value : new Uint8Array()));
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseStoreKVPair);
        if (object.storeKey !== undefined && object.storeKey !== null) {
            message.storeKey = object.storeKey;
        }
        else {
            message.storeKey = "";
        }
        if (object.delete !== undefined && object.delete !== null) {
            message.delete = object.delete;
        }
        else {
            message.delete = false;
        }
        if (object.key !== undefined && object.key !== null) {
            message.key = object.key;
        }
        else {
            message.key = new Uint8Array();
        }
        if (object.value !== undefined && object.value !== null) {
            message.value = object.value;
        }
        else {
            message.value = new Uint8Array();
        }
        return message;
    },
};
const baseBlockMetadata = {};
exports.BlockMetadata = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.requestBeginBlock !== undefined) {
            types_1.RequestBeginBlock.encode(message.requestBeginBlock, writer.uint32(10).fork()).ldelim();
        }
        if (message.responseBeginBlock !== undefined) {
            types_1.ResponseBeginBlock.encode(message.responseBeginBlock, writer.uint32(18).fork()).ldelim();
        }
        for (const v of message.deliverTxs) {
            exports.BlockMetadata_DeliverTx.encode(v, writer.uint32(26).fork()).ldelim();
        }
        if (message.requestEndBlock !== undefined) {
            types_1.RequestEndBlock.encode(message.requestEndBlock, writer.uint32(34).fork()).ldelim();
        }
        if (message.responseEndBlock !== undefined) {
            types_1.ResponseEndBlock.encode(message.responseEndBlock, writer.uint32(42).fork()).ldelim();
        }
        if (message.responseCommit !== undefined) {
            types_1.ResponseCommit.encode(message.responseCommit, writer.uint32(50).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseBlockMetadata);
        message.deliverTxs = [];
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.requestBeginBlock = types_1.RequestBeginBlock.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.responseBeginBlock = types_1.ResponseBeginBlock.decode(reader, reader.uint32());
                    break;
                case 3:
                    message.deliverTxs.push(exports.BlockMetadata_DeliverTx.decode(reader, reader.uint32()));
                    break;
                case 4:
                    message.requestEndBlock = types_1.RequestEndBlock.decode(reader, reader.uint32());
                    break;
                case 5:
                    message.responseEndBlock = types_1.ResponseEndBlock.decode(reader, reader.uint32());
                    break;
                case 6:
                    message.responseCommit = types_1.ResponseCommit.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseBlockMetadata);
        message.deliverTxs = [];
        if (object.requestBeginBlock !== undefined && object.requestBeginBlock !== null) {
            message.requestBeginBlock = types_1.RequestBeginBlock.fromJSON(object.requestBeginBlock);
        }
        else {
            message.requestBeginBlock = undefined;
        }
        if (object.responseBeginBlock !== undefined && object.responseBeginBlock !== null) {
            message.responseBeginBlock = types_1.ResponseBeginBlock.fromJSON(object.responseBeginBlock);
        }
        else {
            message.responseBeginBlock = undefined;
        }
        if (object.deliverTxs !== undefined && object.deliverTxs !== null) {
            for (const e of object.deliverTxs) {
                message.deliverTxs.push(exports.BlockMetadata_DeliverTx.fromJSON(e));
            }
        }
        if (object.requestEndBlock !== undefined && object.requestEndBlock !== null) {
            message.requestEndBlock = types_1.RequestEndBlock.fromJSON(object.requestEndBlock);
        }
        else {
            message.requestEndBlock = undefined;
        }
        if (object.responseEndBlock !== undefined && object.responseEndBlock !== null) {
            message.responseEndBlock = types_1.ResponseEndBlock.fromJSON(object.responseEndBlock);
        }
        else {
            message.responseEndBlock = undefined;
        }
        if (object.responseCommit !== undefined && object.responseCommit !== null) {
            message.responseCommit = types_1.ResponseCommit.fromJSON(object.responseCommit);
        }
        else {
            message.responseCommit = undefined;
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.requestBeginBlock !== undefined &&
            (obj.requestBeginBlock = message.requestBeginBlock
                ? types_1.RequestBeginBlock.toJSON(message.requestBeginBlock)
                : undefined);
        message.responseBeginBlock !== undefined &&
            (obj.responseBeginBlock = message.responseBeginBlock
                ? types_1.ResponseBeginBlock.toJSON(message.responseBeginBlock)
                : undefined);
        if (message.deliverTxs) {
            obj.deliverTxs = message.deliverTxs.map((e) => (e ? exports.BlockMetadata_DeliverTx.toJSON(e) : undefined));
        }
        else {
            obj.deliverTxs = [];
        }
        message.requestEndBlock !== undefined &&
            (obj.requestEndBlock = message.requestEndBlock
                ? types_1.RequestEndBlock.toJSON(message.requestEndBlock)
                : undefined);
        message.responseEndBlock !== undefined &&
            (obj.responseEndBlock = message.responseEndBlock
                ? types_1.ResponseEndBlock.toJSON(message.responseEndBlock)
                : undefined);
        message.responseCommit !== undefined &&
            (obj.responseCommit = message.responseCommit
                ? types_1.ResponseCommit.toJSON(message.responseCommit)
                : undefined);
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseBlockMetadata);
        message.deliverTxs = [];
        if (object.requestBeginBlock !== undefined && object.requestBeginBlock !== null) {
            message.requestBeginBlock = types_1.RequestBeginBlock.fromPartial(object.requestBeginBlock);
        }
        else {
            message.requestBeginBlock = undefined;
        }
        if (object.responseBeginBlock !== undefined && object.responseBeginBlock !== null) {
            message.responseBeginBlock = types_1.ResponseBeginBlock.fromPartial(object.responseBeginBlock);
        }
        else {
            message.responseBeginBlock = undefined;
        }
        if (object.deliverTxs !== undefined && object.deliverTxs !== null) {
            for (const e of object.deliverTxs) {
                message.deliverTxs.push(exports.BlockMetadata_DeliverTx.fromPartial(e));
            }
        }
        if (object.requestEndBlock !== undefined && object.requestEndBlock !== null) {
            message.requestEndBlock = types_1.RequestEndBlock.fromPartial(object.requestEndBlock);
        }
        else {
            message.requestEndBlock = undefined;
        }
        if (object.responseEndBlock !== undefined && object.responseEndBlock !== null) {
            message.responseEndBlock = types_1.ResponseEndBlock.fromPartial(object.responseEndBlock);
        }
        else {
            message.responseEndBlock = undefined;
        }
        if (object.responseCommit !== undefined && object.responseCommit !== null) {
            message.responseCommit = types_1.ResponseCommit.fromPartial(object.responseCommit);
        }
        else {
            message.responseCommit = undefined;
        }
        return message;
    },
};
const baseBlockMetadata_DeliverTx = {};
exports.BlockMetadata_DeliverTx = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.request !== undefined) {
            types_1.RequestDeliverTx.encode(message.request, writer.uint32(10).fork()).ldelim();
        }
        if (message.response !== undefined) {
            types_1.ResponseDeliverTx.encode(message.response, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseBlockMetadata_DeliverTx);
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.request = types_1.RequestDeliverTx.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.response = types_1.ResponseDeliverTx.decode(reader, reader.uint32());
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseBlockMetadata_DeliverTx);
        if (object.request !== undefined && object.request !== null) {
            message.request = types_1.RequestDeliverTx.fromJSON(object.request);
        }
        else {
            message.request = undefined;
        }
        if (object.response !== undefined && object.response !== null) {
            message.response = types_1.ResponseDeliverTx.fromJSON(object.response);
        }
        else {
            message.response = undefined;
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.request !== undefined &&
            (obj.request = message.request ? types_1.RequestDeliverTx.toJSON(message.request) : undefined);
        message.response !== undefined &&
            (obj.response = message.response ? types_1.ResponseDeliverTx.toJSON(message.response) : undefined);
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseBlockMetadata_DeliverTx);
        if (object.request !== undefined && object.request !== null) {
            message.request = types_1.RequestDeliverTx.fromPartial(object.request);
        }
        else {
            message.request = undefined;
        }
        if (object.response !== undefined && object.response !== null) {
            message.response = types_1.ResponseDeliverTx.fromPartial(object.response);
        }
        else {
            message.response = undefined;
        }
        return message;
    },
};
var globalThis = (() => {
    if (typeof globalThis !== "undefined")
        return globalThis;
    if (typeof self !== "undefined")
        return self;
    if (typeof window !== "undefined")
        return window;
    if (typeof global !== "undefined")
        return global;
    throw "Unable to locate global object";
})();
const atob = globalThis.atob || ((b64) => globalThis.Buffer.from(b64, "base64").toString("binary"));
function bytesFromBase64(b64) {
    const bin = atob(b64);
    const arr = new Uint8Array(bin.length);
    for (let i = 0; i < bin.length; ++i) {
        arr[i] = bin.charCodeAt(i);
    }
    return arr;
}
const btoa = globalThis.btoa || ((bin) => globalThis.Buffer.from(bin, "binary").toString("base64"));
function base64FromBytes(arr) {
    const bin = [];
    for (const byte of arr) {
        bin.push(String.fromCharCode(byte));
    }
    return btoa(bin.join(""));
}
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
