// @terra-money/terra.proto/stride/stakeibc/genesis.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenesisState = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const params_1 = require("../../stride/stakeibc/params");
const host_zone_1 = require("../../stride/stakeibc/host_zone");
const epoch_tracker_1 = require("../../stride/stakeibc/epoch_tracker");
exports.protobufPackage = "stride.stakeibc";
const baseGenesisState = { portId: "" };
exports.GenesisState = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.params !== undefined) {
            params_1.Params.encode(message.params, writer.uint32(10).fork()).ldelim();
        }
        if (message.portId !== "") {
            writer.uint32(18).string(message.portId);
        }
        for (const v of message.hostZoneList) {
            host_zone_1.HostZone.encode(v, writer.uint32(42).fork()).ldelim();
        }
        for (const v of message.epochTrackerList) {
            epoch_tracker_1.EpochTracker.encode(v, writer.uint32(82).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseGenesisState);
        message.hostZoneList = [];
        message.epochTrackerList = [];
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.params = params_1.Params.decode(reader, reader.uint32());
                    break;
                case 2:
                    message.portId = reader.string();
                    break;
                case 5:
                    message.hostZoneList.push(host_zone_1.HostZone.decode(reader, reader.uint32()));
                    break;
                case 10:
                    message.epochTrackerList.push(epoch_tracker_1.EpochTracker.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseGenesisState);
        message.hostZoneList = [];
        message.epochTrackerList = [];
        if (object.params !== undefined && object.params !== null) {
            message.params = params_1.Params.fromJSON(object.params);
        }
        else {
            message.params = undefined;
        }
        if (object.portId !== undefined && object.portId !== null) {
            message.portId = String(object.portId);
        }
        else {
            message.portId = "";
        }
        if (object.hostZoneList !== undefined && object.hostZoneList !== null) {
            for (const e of object.hostZoneList) {
                message.hostZoneList.push(host_zone_1.HostZone.fromJSON(e));
            }
        }
        if (object.epochTrackerList !== undefined && object.epochTrackerList !== null) {
            for (const e of object.epochTrackerList) {
                message.epochTrackerList.push(epoch_tracker_1.EpochTracker.fromJSON(e));
            }
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.params !== undefined && (obj.params = message.params ? params_1.Params.toJSON(message.params) : undefined);
        message.portId !== undefined && (obj.portId = message.portId);
        if (message.hostZoneList) {
            obj.hostZoneList = message.hostZoneList.map((e) => (e ? host_zone_1.HostZone.toJSON(e) : undefined));
        }
        else {
            obj.hostZoneList = [];
        }
        if (message.epochTrackerList) {
            obj.epochTrackerList = message.epochTrackerList.map((e) => (e ? epoch_tracker_1.EpochTracker.toJSON(e) : undefined));
        }
        else {
            obj.epochTrackerList = [];
        }
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseGenesisState);
        message.hostZoneList = [];
        message.epochTrackerList = [];
        if (object.params !== undefined && object.params !== null) {
            message.params = params_1.Params.fromPartial(object.params);
        }
        else {
            message.params = undefined;
        }
        if (object.portId !== undefined && object.portId !== null) {
            message.portId = object.portId;
        }
        else {
            message.portId = "";
        }
        if (object.hostZoneList !== undefined && object.hostZoneList !== null) {
            for (const e of object.hostZoneList) {
                message.hostZoneList.push(host_zone_1.HostZone.fromPartial(e));
            }
        }
        if (object.epochTrackerList !== undefined && object.epochTrackerList !== null) {
            for (const e of object.epochTrackerList) {
                message.epochTrackerList.push(epoch_tracker_1.EpochTracker.fromPartial(e));
            }
        }
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
