// @terra-money/terra.proto/stride/stakeibc/ica_account.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ICAAccount = exports.iCAAccountTypeToJSON = exports.iCAAccountTypeFromJSON = exports.ICAAccountType = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "stride.stakeibc";
var ICAAccountType;
(function (ICAAccountType) {
    ICAAccountType[ICAAccountType["DELEGATION"] = 0] = "DELEGATION";
    ICAAccountType[ICAAccountType["FEE"] = 1] = "FEE";
    ICAAccountType[ICAAccountType["WITHDRAWAL"] = 2] = "WITHDRAWAL";
    ICAAccountType[ICAAccountType["REDEMPTION"] = 3] = "REDEMPTION";
    ICAAccountType[ICAAccountType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ICAAccountType = exports.ICAAccountType || (exports.ICAAccountType = {}));
function iCAAccountTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "DELEGATION":
            return ICAAccountType.DELEGATION;
        case 1:
        case "FEE":
            return ICAAccountType.FEE;
        case 2:
        case "WITHDRAWAL":
            return ICAAccountType.WITHDRAWAL;
        case 3:
        case "REDEMPTION":
            return ICAAccountType.REDEMPTION;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ICAAccountType.UNRECOGNIZED;
    }
}
exports.iCAAccountTypeFromJSON = iCAAccountTypeFromJSON;
function iCAAccountTypeToJSON(object) {
    switch (object) {
        case ICAAccountType.DELEGATION:
            return "DELEGATION";
        case ICAAccountType.FEE:
            return "FEE";
        case ICAAccountType.WITHDRAWAL:
            return "WITHDRAWAL";
        case ICAAccountType.REDEMPTION:
            return "REDEMPTION";
        default:
            return "UNKNOWN";
    }
}
exports.iCAAccountTypeToJSON = iCAAccountTypeToJSON;
const baseICAAccount = { address: "", target: 0 };
exports.ICAAccount = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        if (message.target !== 0) {
            writer.uint32(24).int32(message.target);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseICAAccount);
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.address = reader.string();
                    break;
                case 3:
                    message.target = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseICAAccount);
        if (object.address !== undefined && object.address !== null) {
            message.address = String(object.address);
        }
        else {
            message.address = "";
        }
        if (object.target !== undefined && object.target !== null) {
            message.target = iCAAccountTypeFromJSON(object.target);
        }
        else {
            message.target = 0;
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.address !== undefined && (obj.address = message.address);
        message.target !== undefined && (obj.target = iCAAccountTypeToJSON(message.target));
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseICAAccount);
        if (object.address !== undefined && object.address !== null) {
            message.address = object.address;
        }
        else {
            message.address = "";
        }
        if (object.target !== undefined && object.target !== null) {
            message.target = object.target;
        }
        else {
            message.target = 0;
        }
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
