// @terra-money/terra.proto/stride/stakeibc/min_validator_requirements.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MinValidatorRequirements = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const minimal_1 = __importDefault(require("protobufjs/minimal"));
exports.protobufPackage = "stride.stakeibc";
const baseMinValidatorRequirements = { commissionRate: 0, uptime: 0 };
exports.MinValidatorRequirements = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        if (message.commissionRate !== 0) {
            writer.uint32(8).int32(message.commissionRate);
        }
        if (message.uptime !== 0) {
            writer.uint32(16).int32(message.uptime);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseMinValidatorRequirements);
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.commissionRate = reader.int32();
                    break;
                case 2:
                    message.uptime = reader.int32();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseMinValidatorRequirements);
        if (object.commissionRate !== undefined && object.commissionRate !== null) {
            message.commissionRate = Number(object.commissionRate);
        }
        else {
            message.commissionRate = 0;
        }
        if (object.uptime !== undefined && object.uptime !== null) {
            message.uptime = Number(object.uptime);
        }
        else {
            message.uptime = 0;
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        message.commissionRate !== undefined && (obj.commissionRate = message.commissionRate);
        message.uptime !== undefined && (obj.uptime = message.uptime);
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseMinValidatorRequirements);
        if (object.commissionRate !== undefined && object.commissionRate !== null) {
            message.commissionRate = object.commissionRate;
        }
        else {
            message.commissionRate = 0;
        }
        if (object.uptime !== undefined && object.uptime !== null) {
            message.uptime = object.uptime;
        }
        else {
            message.uptime = 0;
        }
        return message;
    },
};
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
