// @opzlab/terrain/lib/commands/contract/instantiate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const command_1 = require("@oclif/command");
const dedent_1 = tslib_1.__importDefault(require("dedent"));
const feather_js_1 = require("@terra-money/feather.js");
const config_1 = require("../../config");
const deployment_1 = require("../../lib/deployment");
const signer_1 = require("../../lib/signer");
const flag = tslib_1.__importStar(require("../../lib/flag"));
const runCommand_1 = tslib_1.__importDefault(require("../../lib/runCommand"));
const defaultErrorCheck_1 = tslib_1.__importDefault(require("../../lib/defaultErrorCheck"));
const TerrainCLI_1 = tslib_1.__importDefault(require("../../TerrainCLI"));
class ContractInstantiate extends command_1.Command {
    async run() {
        const { args, flags } = this.parse(ContractInstantiate);
        let contractAddress;
        let admin;
        // Command execution path.
        const command = async () => {
            const connections = (0, config_1.loadConnections)(flags.prefix);
            const config = (0, config_1.loadConfig)();
            const conf = config(flags.network, args.contract);
            const connection = connections(flags.network);
            const lcd = new feather_js_1.LCDClient({ [connection.chainID]: connection });
            const signer = await (0, signer_1.getSigner)({
                network: flags.network,
                signerId: flags.signer,
                keysPath: flags['keys-path'],
                lcd,
                prefix: flags.prefix,
            });
            admin = signer.key.accAddress(flags.prefix);
            contractAddress = await (0, deployment_1.instantiate)({
                conf,
                signer,
                admin,
                contract: args.contract,
                codeId: flags['code-id'],
                network: flags.network,
                instanceId: flags['instance-id'],
                refsPath: flags['refs-path'],
                lcd,
                prefix: flags.prefix,
            });
        };
        // Message to be displayed upon successful command execution.
        const successMessage = () => {
            TerrainCLI_1.default.success((0, dedent_1.default) `
        Contract "${args.contract}" was successfully instantiated on "${flags.network}".\n
        Contract Address: "${contractAddress}"\n
        Administrator: "${admin}"
      `, 'Contract Instantiated');
        };
        // Attempt to execute command while backtracking through file tree.
        await (0, runCommand_1.default)(config_1.CONFIG_FILE_NAME, command, (0, defaultErrorCheck_1.default)(args.contract), successMessage);
    }
}
ContractInstantiate.description = 'Instantiate the contract.';
ContractInstantiate.flags = Object.assign(Object.assign({ 'instance-id': flag.instanceId, 'code-id': flag.codeId }, flag.tx), flag.terrainPaths);
ContractInstantiate.args = [{ name: 'contract', required: true }];
exports.default = ContractInstantiate;
