// @terra-money/feather.js/dist/core/staking/msgs/MsgCancelUnbondingDelegation.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MsgCancelUnbondingDelegation = void 0;
var json_1 = require("../../../util/json");
var Coin_1 = require("../../Coin");
var any_1 = require("@terra-money/terra.proto/google/protobuf/any");
var tx_1 = require("@terra-money/terra.proto/cosmos/staking/v1beta1/tx");
/**
 * When a user wants to cancel the unbonding this messaage can be used
 */
var MsgCancelUnbondingDelegation = /** @class */ (function (_super) {
    __extends(MsgCancelUnbondingDelegation, _super);
    /**
     *
     * @param delegator_address delegator's account address
     * @param validator_src_address source validator to cancel the undelegation from
     * @param validator_dst_address destination validator to cancel the undelegation from
     * @param amount amount of luna to cancel the undelegation from
     */
    function MsgCancelUnbondingDelegation(delegator_address, validator_address, creation_height, amount) {
        var _this = _super.call(this) || this;
        _this.delegator_address = delegator_address;
        _this.validator_address = validator_address;
        _this.creation_height = creation_height;
        _this.amount = amount;
        return _this;
    }
    MsgCancelUnbondingDelegation.fromAmino = function (data, _) {
        _;
        var _a = data.value, delegator_address = _a.delegator_address, validator_address = _a.validator_address, creation_height = _a.creation_height, amount = _a.amount;
        return new MsgCancelUnbondingDelegation(delegator_address, validator_address, creation_height, amount ? Coin_1.Coin.fromAmino(amount) : undefined);
    };
    MsgCancelUnbondingDelegation.prototype.toAmino = function (isClassic) {
        var _a = this, delegator_address = _a.delegator_address, validator_address = _a.validator_address, creation_height = _a.creation_height, amount = _a.amount;
        return {
            type: isClassic
                ? 'staking/MsgCancelUnbondingDelegation'
                : 'cosmos-sdk/MsgCancelUnbondingDelegation',
            value: {
                delegator_address: delegator_address,
                validator_address: validator_address,
                creation_height: creation_height,
                amount: amount === null || amount === void 0 ? void 0 : amount.toAmino(),
            },
        };
    };
    MsgCancelUnbondingDelegation.fromData = function (data, _) {
        _;
        var delegator_address = data.delegator_address, validator_address = data.validator_address, creation_height = data.creation_height, amount = data.amount;
        return new MsgCancelUnbondingDelegation(delegator_address, validator_address, creation_height, amount ? Coin_1.Coin.fromData(amount) : undefined);
    };
    MsgCancelUnbondingDelegation.prototype.toData = function (_) {
        _;
        var _a = this, delegator_address = _a.delegator_address, validator_address = _a.validator_address, creation_height = _a.creation_height, amount = _a.amount;
        return {
            '@type': '/cosmos.staking.v1beta1.MsgCancelUnbondingDelegation',
            delegator_address: delegator_address,
            validator_address: validator_address,
            creation_height: creation_height,
            amount: amount === null || amount === void 0 ? void 0 : amount.toData(),
        };
    };
    MsgCancelUnbondingDelegation.fromProto = function (proto, _) {
        _;
        return new MsgCancelUnbondingDelegation(proto.delegatorAddress, proto.validatorAddress, proto.creationHeight, Coin_1.Coin.fromProto(proto.amount));
    };
    MsgCancelUnbondingDelegation.prototype.toProto = function (_) {
        _;
        var _a = this, delegator_address = _a.delegator_address, validator_address = _a.validator_address, creation_height = _a.creation_height, amount = _a.amount;
        return tx_1.MsgCancelUnbondingDelegation.fromPartial({
            delegatorAddress: delegator_address,
            validatorAddress: validator_address,
            creationHeight: creation_height,
            amount: amount === null || amount === void 0 ? void 0 : amount.toProto(),
        });
    };
    MsgCancelUnbondingDelegation.prototype.packAny = function (isClassic) {
        return any_1.Any.fromPartial({
            typeUrl: '/cosmos.staking.v1beta1.MsgCancelUnbondingDelegation',
            value: tx_1.MsgCancelUnbondingDelegation.encode(this.toProto(isClassic)).finish(),
        });
    };
    MsgCancelUnbondingDelegation.unpackAny = function (msgAny, isClassic) {
        return MsgCancelUnbondingDelegation.fromProto(tx_1.MsgCancelUnbondingDelegation.decode(msgAny.value), isClassic);
    };
    return MsgCancelUnbondingDelegation;
}(json_1.JSONSerializable));
exports.MsgCancelUnbondingDelegation = MsgCancelUnbondingDelegation;
