// @terra-money/terra.proto/stride/interchainquery/v1/query.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcWebImpl = exports.QueryServicePendingQueriesDesc = exports.QueryServiceDesc = exports.QueryServiceClientImpl = exports.QueryPendingQueriesResponse = exports.QueryPendingQueriesRequest = exports.protobufPackage = void 0;
/* eslint-disable */
const long_1 = __importDefault(require("long"));
const grpc_web_1 = require("@improbable-eng/grpc-web");
const minimal_1 = __importDefault(require("protobufjs/minimal"));
const browser_headers_1 = require("browser-headers");
const genesis_1 = require("../../../stride/interchainquery/v1/genesis");
exports.protobufPackage = "stride.interchainquery.v1";
const baseQueryPendingQueriesRequest = {};
exports.QueryPendingQueriesRequest = {
    encode(_, writer = minimal_1.default.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseQueryPendingQueriesRequest);
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(_) {
        const message = Object.assign({}, baseQueryPendingQueriesRequest);
        return message;
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    fromPartial(_) {
        const message = Object.assign({}, baseQueryPendingQueriesRequest);
        return message;
    },
};
const baseQueryPendingQueriesResponse = {};
exports.QueryPendingQueriesResponse = {
    encode(message, writer = minimal_1.default.Writer.create()) {
        for (const v of message.pendingQueries) {
            genesis_1.Query.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof minimal_1.default.Reader ? input : new minimal_1.default.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = Object.assign({}, baseQueryPendingQueriesResponse);
        message.pendingQueries = [];
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.pendingQueries.push(genesis_1.Query.decode(reader, reader.uint32()));
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        const message = Object.assign({}, baseQueryPendingQueriesResponse);
        message.pendingQueries = [];
        if (object.pendingQueries !== undefined && object.pendingQueries !== null) {
            for (const e of object.pendingQueries) {
                message.pendingQueries.push(genesis_1.Query.fromJSON(e));
            }
        }
        return message;
    },
    toJSON(message) {
        const obj = {};
        if (message.pendingQueries) {
            obj.pendingQueries = message.pendingQueries.map((e) => (e ? genesis_1.Query.toJSON(e) : undefined));
        }
        else {
            obj.pendingQueries = [];
        }
        return obj;
    },
    fromPartial(object) {
        const message = Object.assign({}, baseQueryPendingQueriesResponse);
        message.pendingQueries = [];
        if (object.pendingQueries !== undefined && object.pendingQueries !== null) {
            for (const e of object.pendingQueries) {
                message.pendingQueries.push(genesis_1.Query.fromPartial(e));
            }
        }
        return message;
    },
};
class QueryServiceClientImpl {
    constructor(rpc) {
        this.rpc = rpc;
        this.PendingQueries = this.PendingQueries.bind(this);
    }
    PendingQueries(request, metadata) {
        return this.rpc.unary(exports.QueryServicePendingQueriesDesc, exports.QueryPendingQueriesRequest.fromPartial(request), metadata);
    }
}
exports.QueryServiceClientImpl = QueryServiceClientImpl;
exports.QueryServiceDesc = {
    serviceName: "stride.interchainquery.v1.QueryService",
};
exports.QueryServicePendingQueriesDesc = {
    methodName: "PendingQueries",
    service: exports.QueryServiceDesc,
    requestStream: false,
    responseStream: false,
    requestType: {
        serializeBinary() {
            return exports.QueryPendingQueriesRequest.encode(this).finish();
        },
    },
    responseType: {
        deserializeBinary(data) {
            return Object.assign(Object.assign({}, exports.QueryPendingQueriesResponse.decode(data)), { toObject() {
                    return this;
                } });
        },
    },
};
class GrpcWebImpl {
    constructor(host, options) {
        this.host = host;
        this.options = options;
    }
    unary(methodDesc, _request, metadata) {
        var _a;
        const request = Object.assign(Object.assign({}, _request), methodDesc.requestType);
        const maybeCombinedMetadata = metadata && this.options.metadata
            ? new browser_headers_1.BrowserHeaders(Object.assign(Object.assign({}, (_a = this.options) === null || _a === void 0 ? void 0 : _a.metadata.headersMap), metadata === null || metadata === void 0 ? void 0 : metadata.headersMap))
            : metadata || this.options.metadata;
        return new Promise((resolve, reject) => {
            grpc_web_1.grpc.unary(methodDesc, {
                request,
                host: this.host,
                metadata: maybeCombinedMetadata,
                transport: this.options.transport,
                debug: this.options.debug,
                onEnd: function (response) {
                    if (response.status === grpc_web_1.grpc.Code.OK) {
                        resolve(response.message);
                    }
                    else {
                        const err = new Error(response.statusMessage);
                        err.code = response.status;
                        err.metadata = response.trailers;
                        reject(err);
                    }
                },
            });
        });
    }
}
exports.GrpcWebImpl = GrpcWebImpl;
if (minimal_1.default.util.Long !== long_1.default) {
    minimal_1.default.util.Long = long_1.default;
    minimal_1.default.configure();
}
